routing/router.go

```diff
@@ -451,6 +474,42 @@ func (r *ChannelRouter) Start() error {
 		return err
 	}
 
+	// If any payments are still active in the payment state machine, we
+	// resume, to make sure their results are properly handled.
+	payments, err := r.paymentStates.fetchActivePayments()
+	if err != nil {
+		return err
+	}
+
+	for _, paymentHash := range payments {
+		log.Infof("Resuming payment with hash %x", paymentHash)
+
+		r.wg.Add(1)
+		go func(paymentHash lntypes.Hash) {
+			defer r.wg.Done()
+
+			// We create a dummy, empty payment session such that
+			// we won't attempt another payment attempt when the
+			// result for this payment is received.
+			paySession := r.missionControl.NewPaymentSessionFromRoutes(
+				nil,
+			)
+
+			payment := &LightningPayment{
+				PaymentHash: paymentHash,
+			}
```

Comment:
What happens to other properties of `LightningPayment` that we need on resume? Fee limit, route hints, etc. If we are not past the timeout, we need those to regenerate routes for further attempts.

Reply:
We only need the payment hash for now, as the payment session won't produce more routes. Storing this info is left for another PR.

Reply:
What happens in the current implementation when a payment is resumed after restart, it fails within  the payment session timeout and a new route is requested? That would need all the routing parameters?

Reply:
Since we pass `nil` to `NewPaymentSessionFromRoutes` above, no more routes will be available, and it will fail after the result is back. Storing the information to re-create the payment session is left to later PR.

Reply:
Add todo comment?

Reply:
Shouldn't we only resume if the payment doesn't have a new attempt? Otherwise, it's possible for us to send out a duplicate payment with the same payment hash.

Reply:
Resume in this context is resuming the payment loop, which will first check whether an attempt is already in flight, and query the switch for a result, before potentially making a new attempt (which by design will fail until we store all information needed to restore the payment session).